src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -69,6 +69,8 @@ export class CommandDetectionCapability implements ICommandDetectionCapability {
 	readonly onCommandStarted = this._onCommandStarted.event;
 	private readonly _onCommandFinished = new Emitter<ITerminalCommand>();
 	readonly onCommandFinished = this._onCommandFinished.event;
+	private readonly _onCommandCleared = new Emitter<ITerminalCommand[]>();
+	readonly onCommandCleared = this._onCommandCleared.event;
```

Comment:
I think invalidated is more accurate here as this will be useful on Windows when commands essentially become broken under certain circumstances.